File: DarkOrbit 10.0/Net/netty/commands/AsteroidProgressCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ow.Utils;

namespace Ow.Net.netty.commands
{
    class AsteroidProgressCommand
    {
        public const short ID = 1995;

        public String bestProgressClanName = "";     
        public float bestProgress = 0;     
        public Boolean buildButtonActive = false;      
        public String ownClanName = "";     
        public EquippedModulesModule state;     
        public int battleStationId = 0;      
        public float ownProgress = 0;

        public AsteroidProgressCommand(int battleStationId, float ownProgress, float bestProgress, string ownClanName, string bestProgressClanName, EquippedModulesModule state, bool buildButtonActive)
        {
            this.battleStationId = battleStationId;
            this.ownProgress = ownProgress;
            this.bestProgress = bestProgress;
            this.ownClanName = ownClanName;
            this.bestProgressClanName = bestProgressClanName;
            this.state = state;
            this.buildButtonActive = buildButtonActive;
        }

        public byte[] write()
        {
            var param1 = new ByteArray(ID);
            param1.writeUTF(bestProgressClanName);
            param1.writeFloat(bestProgress);
            param1.writeBoolean(buildButtonActive);
            param1.writeUTF(ownClanName);
            param1.write(state.write());
            param1.writeInt(battleStationId << 9 | battleStationId >> 23);
            param1.writeFloat(ownProgress);
            return param1.Message.ToArray();
        }
    }
}

File: DarkOrbit 9.0/DarkOrbit 9.0 Emulator/src/simulator/netty/serverCommands/AsteroidProgressCommand.java
package simulator.netty.serverCommands;

import java.io.DataOutputStream;
import java.io.IOException;

/**
 Created by Pedro on 22-03-2015.
 */
public class AsteroidProgressCommand {

    public static int ID = 12826;

    public boolean               buildButtonActive;
    public long                  bestProgress;
    public String                ownClanName;
    public EquippedModulesModule state;
    public long                  ownProgress;
    public int                   battleStationId;
    public String                bestProgressClanName;

    public AsteroidProgressCommand(int param1, long param2, long param3, String param4, String param5,
                                   EquippedModulesModule param6, boolean param7) {
        this.battleStationId = param1;
        this.ownProgress = param2;
        this.bestProgress = param3;
        this.ownClanName = param4;
        this.bestProgressClanName = param5;
        this.state = param6;
        this.buildButtonActive = param7;
    }

    public void write(DataOutputStream param1) {
        try {
            param1.writeShort(ID);
            this.writeInternal(param1);
        } catch (IOException e) {
        }
    }

    protected void writeInternal(DataOutputStream param1) {
        try {
            param1.writeBoolean(this.buildButtonActive);
            param1.writeFloat(this.bestProgress);
            param1.writeShort(31634);
            param1.writeUTF(this.ownClanName);
            this.state.write(param1);
            param1.writeFloat(this.ownProgress);
            param1.writeInt(this.battleStationId << 14 | this.battleStationId >>> 18);
            param1.writeUTF(this.bestProgressClanName);
        } catch (IOException e) {
        }
    }
}
